libs/langchain-azure-cosmosdb/src/chat_histories/mongodb.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  Collection,
+  Document as AzureCosmosMongoDBDocument,
+  PushOperator,
+  Db,
+  MongoClient,
+} from "mongodb";
+import { BaseListChatMessageHistory } from "@langchain/core/chat_history";
+import {
+  BaseMessage,
+  mapChatMessagesToStoredMessages,
+  mapStoredMessagesToChatMessages,
+} from "@langchain/core/messages";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+
+export interface AzureCosmosDBMongoChatHistoryDBConfig {
+  readonly client?: MongoClient;
+  readonly connectionString?: string;
+  readonly databaseName?: string;
+  readonly collectionName?: string;
+}
+
+export class AzureCosmosDBMongoChatMessageHistory extends BaseListChatMessageHistory {
+  lc_namespace = ["langchain", "stores", "message", "azurecosmosdb"];
+
+  get lc_secrets(): { [key: string]: string } {
+    return {
+      connectionString: "AZURE_COSMOSDB_MONGODB_CONNECTION_STRING",
+    };
+  }
+
+  private initPromise?: Promise<void>;
+
+  private readonly client: MongoClient | undefined;
+
+  private database: Db;
+
+  private collection: Collection<AzureCosmosMongoDBDocument>;
+
+  private sessionId: string;
+
+  private idKey = "sessionId";
```

Comment:
nit: could be a constant outside the class since it's not configurable

Reply:
resolved